interceptor.go

```diff
@@ -48,65 +49,20 @@ func New(opts *Opts) Interceptor {
 	return ic
 }
 
-func (ic *interceptor) Intercept(w http.ResponseWriter, req *http.Request, forwardInitialRequest bool, op ops.Op, defaultPort int) {
-	addr := hostIncludingPort(req, defaultPort)
-	port, err := portForAddress(addr)
-	if err != nil {
-		respondBadGateway(w, op.FailIf(errors.New("Unable to determine port for address %v: %v", addr, err)))
-		return
-	}
-
-	var downstream net.Conn
-	var downstreamBuffered *bufio.ReadWriter
-	var upstream net.Conn
-
-	closeDownstream := false
-	closeUpstream := false
-	defer func() {
-		if closeDownstream {
-			if closeErr := downstream.Close(); closeErr != nil {
-				log.Tracef("Error closing downstream connection: %s", closeErr)
-			}
-		}
-		if closeUpstream {
-			if closeErr := upstream.Close(); closeErr != nil {
-				log.Tracef("Error closing upstream connection: %s", closeErr)
-			}
-		}
-	}()
-
-	upstream, pipe, err := ic.Dial(req, addr, port)
-	if err != nil {
-		respondBadGateway(w, op.FailIf(errors.New("Could not dial '%v': %v", addr, err)))
-		return
-	}
-	closeUpstream = true
-
-	// Hijack underlying connection.
-	downstream, downstreamBuffered, err = w.(http.Hijacker).Hijack()
-	if err != nil {
-		respondBadGateway(w, op.FailIf(errors.New("Unable to hijack connection: %s", err)))
-		return
-	}
-	closeDownstream = true
-
-	if pipe {
-		ic.pipe(w, req, forwardInitialRequest, op, downstream, downstreamBuffered, upstream)
-	} else {
-		ic.http(req, forwardInitialRequest, op, downstream, downstreamBuffered, upstream)
-	}
-}
-
 func (ic *interceptor) respondOK(writer io.Writer, req *http.Request, respHeaders http.Header) error {
-	return ic.respondHijacked(writer, req, http.StatusOK, respHeaders)
+	return ic.respondHijacked(writer, req, http.StatusOK, respHeaders, nil)
 }
 
-func (ic *interceptor) respondBadGatewayHijacked(writer io.Writer, req *http.Request) error {
+func (ic *interceptor) respondBadGatewayHijacked(writer io.Writer, req *http.Request, err error) error {
 	log.Debugf("Responding %v", http.StatusBadGateway)
-	return ic.respondHijacked(writer, req, http.StatusBadGateway, make(http.Header))
+	var body []byte
+	if err != nil {
+		body = []byte(hidden.Clean(err.Error()))
```

Comment:
The error may contain some sensitive server-side information, but yeah, it's good for troubleshooting from client logs.